executor/ddl.go

```diff
@@ -340,6 +342,111 @@ func (e *DDLExec) executeCreateDatabase(s *ast.CreateDatabaseStmt) error {
 	return err
 }
 
+func (e *DDLExec) executeFlashbackDatabase(s *ast.FlashBackDatabaseStmt) error {
+	job, schemaInfo, recoversInfo, err := e.getRecoverDBByDBName(s.DBName)
+	if err != nil {
+		return err
+	}
+	dbName := model.NewCIStr(s.DBName)
+	if len(s.NewName) != 0 {
+		dbName = model.NewCIStr(s.NewName)
+	}
+	// Check the DBName was not exists.
+	is := domain.GetDomain(e.ctx).InfoSchema()
+	if is.SchemaExists(dbName) {
+		return infoschema.ErrDatabaseExists.GenWithStackByArgs(dbName)
+	}
+	schemaInfo.Name = dbName
+	recoverSchemaInfo := &ddl.RecoverSchemaInfo{
+		SchemaInfo:    schemaInfo,
+		RecoversInfo:  recoversInfo,
+		DropJobID:     job.ID,
+		SnapshotTS:    job.StartTS,
+		OldSchemaName: s.DBName,
+	}
+	// Call DDL RecoverSchema.
+	err = domain.GetDomain(e.ctx).DDL().RecoverSchema(e.ctx, recoverSchemaInfo)
+	return err
+}
+
+func (e *DDLExec) getRecoverDBByDBName(schemaName string) (*model.Job, *model.DBInfo, []*ddl.RecoverInfo, error) {
```

Comment:
```suggestion
func (e *DDLExec) getRecoverDBByDBName(schemaName string) (*model.Job, *model.DBInfo, []*ddl.RecoverInfo, error) {
```
How about renaming it to `getRecoverDBByName`?